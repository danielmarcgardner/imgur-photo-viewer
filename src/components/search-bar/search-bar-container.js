import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { searchForImages } from '../../redux/actions/thunks';
import SearchBar from './search-bar';

const mapDispatchToProps = dispatch => bindActionCreators({
  searchForImages,
}, dispatch);

export default connect(null, mapDispatchToProps)(SearchBar);
