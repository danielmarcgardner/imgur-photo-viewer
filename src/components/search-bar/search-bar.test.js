import React from 'react';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SearchBar from './search-bar';
import { mapDispatchToProps } from './search-bar-container';

configure({ adapter: new Adapter() });

const searchForImages = jest.fn();

describe('SearchBar', () => {
  it('Renders to the page', () => {
    const component = shallow(
      <SearchBar
        searchForImages={searchForImages}
      />,
    );
    expect(component).toMatchSnapshot();
  });
  it('Renders with correct properties', () => {
    const component = shallow(
      <SearchBar
        searchForImages={searchForImages}
      />,
    );
    expect(component.find('#SearchBar').length).toBe(1);
    expect(component.find('form').length).toBe(1);
    expect(component.find('input').length).toBe(1);
    expect(component.find('button').length).toBe(1);
  });
  it('onSubmit of form calls correct searchForImages with correct value', () => {
    const component = shallow(
      <SearchBar
        searchForImages={searchForImages}
      />,
    );
    component.find('form').simulate('submit', {
      target: {
        InputBar: {
          value: 'bulldog',
        },
      },
      preventDefault: jest.fn(),
    });
    expect(searchForImages).toHaveBeenCalled();
    expect(searchForImages.mock.calls[0][0]).toBe('bulldog');
  });
  it('mapDispatchToProps dispatches the expected action', () => {
    const dispatch = jest.fn();
    const dispatched = mapDispatchToProps(dispatch);
    expect(dispatched).toHaveProperty('searchForImages');
  });
});
