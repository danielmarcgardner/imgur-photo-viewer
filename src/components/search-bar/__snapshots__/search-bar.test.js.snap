// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Renders to the page 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar
    searchForImages={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={[Function]}
>
        <input
                id="InputBar"
                placeholder="Search Imgur for your favorite images"
                type="text"
        />
        <button>
                Search Imgur!
        </button>
</form>,
      "id": "SearchBar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            id="InputBar"
            placeholder="Search Imgur for your favorite images"
            type="text"
/>,
          <button>
            Search Imgur!
</button>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "id": "InputBar",
            "placeholder": "Search Imgur for your favorite images",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Search Imgur!",
          },
          "ref": null,
          "rendered": "Search Imgur!",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
>
          <input
                    id="InputBar"
                    placeholder="Search Imgur for your favorite images"
                    type="text"
          />
          <button>
                    Search Imgur!
          </button>
</form>,
        "id": "SearchBar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              id="InputBar"
              placeholder="Search Imgur for your favorite images"
              type="text"
/>,
            <button>
              Search Imgur!
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "id": "InputBar",
              "placeholder": "Search Imgur for your favorite images",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Search Imgur!",
            },
            "ref": null,
            "rendered": "Search Imgur!",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
